RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronIDProducer.cc

```diff
@@ -146,14 +173,14 @@ void LowPtGsfElectronIDProducer::produce(edm::StreamID, edm::Event& event, const
 //////////////////////////////////////////////////////////////////////////////////////////
 //
 double LowPtGsfElectronIDProducer::eval(
-    const std::string& name, const reco::GsfElectronRef& ele, double rho, float unbiased, float field_z) const {
+  const std::string& name, const edm::Ptr<reco::GsfElectron>& ele, double rho, float unbiased, float field_z) const {
   auto iter = std::find(names_.begin(), names_.end(), name);
   if (iter != names_.end()) {
     int index = std::distance(names_.begin(), iter);
     std::vector<float> inputs;
     if (version_.empty()) {  // Original XML model
       lowptgsfeleid::Features features;
-      features.set(ele, rho);
+      features.set(*ele, rho);
```

Comment:
@slava77 I am required to provide support for an old model (Autumn18) in 10_6_X that no longer exists in master. The method above is defined [here](https://github.com/cms-sw/cmssw/pull/33992/files?file-filters%5B%5D=.cc&file-filters%5B%5D=.h#diff-82e3821f0636c4998a4527ae7af4178b4300933e838c4d80edda49f32816d44eR48-R49), which originally expected a `const reco::GsfElectronRef&`. Similar methods for newer models (see below this comment) have since been updated to expect an arg `const reco::GsfElectron&` and I have done the same for the above method (while preserving the original interface). Anther change is a switch from Ref to Ptr (which is then dereferenced). I believe the changes in this file and also [here](https://github.com/cms-sw/cmssw/pull/33992/files?file-filters%5B%5D=.cc&file-filters%5B%5D=.h#diff-7240b4145e76390809868606eaaaadfa7ce27d6f85ac55d5e99d6a5fb4ba35bdL47-R96) to be the minimal required while being consistent with other methods (e.g. [here](https://github.com/cms-sw/cmssw/pull/33992/files?file-filters%5B%5D=.cc&file-filters%5B%5D=.h#diff-e7fc6e2e8520a28d35f7f2df77a18f0dd885d69e2bcf20f530614935c99b445bR186)) and those in master.